Sources/Optimizely/OptimizelyJSON.swift

```diff
@@ -0,0 +1,142 @@
+    
+    // MARK: - Init
+    
+    init?(payload: String) {
+        do {
+            guard let data = payload.data(using: .utf8),
+                let dict = try JSONSerialization.jsonObject(with: data, options: .allowFragments) as? [String: Any] else {
+                    return nil
+            }
+            self.map = dict
+            self.payload = payload
+        } catch {
+            return nil
+        }
+    }
+    
+    init?(map: [String: Any]) {
+        if !JSONSerialization.isValidJSONObject(map) {
+            return nil
+        }
+        self.map = map
+    }
+    
+    // MARK: - OptimizelyJSON Implementation
+    
+    /// - Returns: The string representation of json
+    public func toString() -> String? {
+        guard let payload = self.payload else {
+            guard let jsonData = try? JSONSerialization.data(withJSONObject: map,
+                                                             options: []),
+                let jsonString = String(data: jsonData, encoding: .utf8) else {
+                    logger.e(.failedToConvertMapToString)
+                    return nil
+            }
+            self.payload = jsonString
+            return jsonString
+        }
+        return payload
+    }
+    
+    /// - Returns: The json dictionary
+    public func toMap() -> [String: Any] {
+        return map
+    }
+    
+    /// Populates the decodable schema passed by the user
+    ///
+    /// - Parameters:
+    ///   - jsonPath: Key path for the value.
+    ///   - schema: Decodable schema to populate.
+    /// - Returns: true if value decoded successfully
+    public func getValue<T: Decodable>(jsonPath: String, schema: UnsafeMutablePointer<T>) -> Bool {
+        func populateDecodableSchema(value: Any) -> Bool {
+            guard JSONSerialization.isValidJSONObject(value) else {
+                // Try and assign value directly to schema
+                if let v = value as? T {
+                    schema.pointee = v
+                    return true
+                }
+                logger.e(.failedToAssignValueToSchema)
+                return false
+            }
+            // Try to decode value into schema
+            guard let jsonData = try? JSONSerialization.data(withJSONObject: value, options: []),
+                let decodedValue = try? JSONDecoder().decode(T.self, from: jsonData) else {
+                    logger.e(.failedToAssignValueToSchema)
+                    return false
+            }
+            schema.pointee = decodedValue
+            return true
+        }
+        return getValue(jsonPath: jsonPath, schemaHandler: populateDecodableSchema(value:))
+    }
+    
+    /// Populates the schema passed by the user
+    ///
+    /// - Parameters:
+    ///   - jsonPath: Key path for the value.
+    ///   - schema: Schema to populate.
+    /// - Returns: true if value decoded successfully
+    public func getValue<T>(jsonPath: String, schema: UnsafeMutablePointer<T>) -> Bool {
```

Comment:
Can we allow "nil" jsonPath same as ""? Nil is more convenient. We can assign default value to nil:
```suggestion
    public func getValue<T>(jsonPath: String? = nil, schema: UnsafeMutablePointer<T>) -> Bool {
```